.cargo/config.toml

```diff
@@ -1,5 +1,58 @@
-# Deployment runtime lints
-[target.'cfg(all(target_arch = "wasm32", not(test)))']
+# Unfortunately we cannot add any lints based on `cfg(test)` nor `cfg(feature = "feat")`: https://github.com/rust-lang/cargo/issues/8170.
+# Hence, we can only define lints that are applied codebase-wide (including tests), and lints that are applied only to runtime code.
+
+# Codebase-wide lints.
+[target.'cfg(all())']
+rustflags = [
```

Comment:
If I remember correctly, we have agreed, that tests should always live in a tests module. We can enforce that with: https://rust-lang.github.io/rust-clippy/master/index.html#tests_outside_test_module

Reply:
I will double-check this, but I was stumbling upon some false positives. Let me check that again.

Reply:
Ok I found out why. Most pallets use the old benchmarking syntax which mixes tests and non tests together, triggering this lint a lot. I will open a new PR that adds the new lint and migrates the old benchmarking logic to the new one, which should also solve the problem with the lints.

Reply:
gl hf in the staking pallet

Reply:
Example of running clippy with the new lint:

```
error: this function marked with #[test] is outside a #[cfg(test)] module
   --> pallets/attestation/src/benchmarking.rs:31:1
    |
31  | / benchmarks! {
32  | |     where_clause {
33  | |         where
34  | |         T: core::fmt::Debug,
...   |
203 | |     )
204 | | }
    | |_^
    |
    = note: move it to a testing module marked with #[cfg(test)]
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#tests_outside_test_module
    = note: `-D clippy::tests-outside-test-module` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::tests_outside_test_module)]`
    = note: this error originates in the macro `$crate::impl_bench_name_tests` which comes from the expansion of the macro `benchmarks` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Slapping a `#[cfg(test)]` above `impl_benchmark_test_suite!`, gives the following:

```
error: no rules expected the token `cfg`
   --> pallets/attestation/src/benchmarking.rs:199:4
    |
199 |     #[cfg(test)]
    |       ^^^ no rules expected this token in macro call
    |
note: while trying to match `skip_meta`
   --> /home/antonio/.cargo/git/checkouts/polkadot-sdk-cff69157b985ed76/e046d3b/substrate/frame/benchmarking/src/v1.rs:321:5
    |
321 |         #[skip_meta]
```

Reply:
I guess you did not run it with `--all-features`?